app/util/route.js

```diff
@@ -0,0 +1,89 @@
+import {equals} from 'ramda'
+import PreactRouter from 'preact-router'
+import {map, reduce} from 'wasmuth'
+import compose from '/util/compose'
+import {set, dispatch, getState} from '/store'
+import {routes} from '/routes'
+
+const {cloneElement} = Preact
+
+/**
+ * Use instaed of preact-router Router component.
+ * All children should be Route components (defined below).
+ * - Add key prop to Route children so that when route changes,
+ *   the current route component will unmount and the new one will mount.
+ * - Add path prop to Route children based on their name props.
+ */
+export const Router = ({children, routes, ...props}) =>
+  <PreactRouter {...props}>
+    {map((route, i) => cloneElement(route, {
+      key: i,
+      path: routes[route.attributes.name].path
+    }), children)}
+  </PreactRouter>
+
+/**
+ * Add preact-router props into the atom state
+ */
+export const Route = compose({
+  componentWillMount () {
+    const currentValues = getState().route
+    const newValues = {
+      args: this.props.matches,
```

Comment:
This doesn't seem to be used yet? Are matches supposed to come from preact-router?

Reply:
yep preact-router adds matches object that includes any of the matched arguments like 
```
path: /user/:id
url: /user/4?search=hi
matches: {id: 4, search: 'hi'}
```